tests/lib/rules/html-indent.js

```diff
@@ -0,0 +1,4214 @@
+/**
+ * @author Toru Nagashima
+ * @copyright 2016 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/html-indent')
+const RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+* Prevents leading spaces in a multiline template literal from appearing in the resulting string
+* @param {string[]} strings The strings in the template literal
+* @returns {string} The template literal, with spaces removed from all lines
+*/
+function unIndent (strings) {
+  const templateValue = strings[0]
+  const lines = templateValue.replace(/^\n/, '').replace(/\n\s*$/, '').split('\n')
+  const lineIndents = lines.filter(line => line.trim()).map(line => line.match(/ */)[0].length)
+  const minLineIndent = Math.min.apply(null, lineIndents)
+
+  return lines.map(line => line.slice(minLineIndent)).join('\n')
+}
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const tester = new RuleTester({
+  parser: 'vue-eslint-parser',
+  parserOptions: {
+    ecmaVersion: 2017,
+    ecmaFeatures: {
+      globalReturn: true
+    }
+  }
+})
+
+tester.run('html-indent', rule, {
+  valid: [
+    // Examples in docs.
+    unIndent`
+      <template>
+          <div class="foo">
+              Hello.
+          </div>
+          <div
+              id="a"
+              class="b"
+              :other-attr="{
+                  aaa: 1,
+                  bbb: 2
+              }"
+              @other-attr2="
+                  foo();
+                  bar();
+              "
+          >
+              {{
```

Comment:
maybe we need some more tests for `{{}}`.

and I didn't see tests for comma, e.g:
```html
<template>
{{
  foo,
  bar,
  qux
}}
</template>
```

Reply:
In technically, mustaches (`{{foo}}`) and directive values (`"foo"` of directives) are completely same for indentation. Both are `VExpressionContainer`. So I don't think we need many tests for mustaches. I added some tests.